src/NBitcoin/ChainedHeader.cs

```diff
@@ -4,6 +4,60 @@
 
 namespace NBitcoin
 {
+    /// <summary>
+    /// Represents the state of the availability of a block.
+    /// </summary>
+    public enum BlockDataAvailabilityState
+    {
+        /// <summary>
+        /// A <see cref="BlockHeader"/> is found and is a valid header.
+        /// </summary>
+        HeaderOnly,
+
+        /// <summary>
+        /// We are interested in downloading the <see cref="Block"/> that is being represented by the current <see cref="BlockHeader"/>. 
+        /// This happens when a header is a part of the chain that can potentially replace our consensus tip 
+        /// because its tip's total chain work is greater comparing to what we currently have.
+        /// </summary>
+        BlockRequired,
+
+        /// <summary>
+        /// The <see cref="Block"/> is downloaded, if the Block property is null the block needs to be asked from store.
+        /// </summary>
+        BlockAvailable
+    }
+
+    /// <summary>
+    /// Represents the validation level of a block.
+    /// </summary>
+    public enum ValidationState
+    {
+        /// <summary>
+        /// No validation was performed.
+        /// </summary>
+        Unknown,
+
+        /// <summary>
+        /// We have a valid block header.
+        /// </summary>
+        HeaderValidated,
+
+        /// <summary>
+        /// No validation required because the block is assumed to be valid.
+        /// </summary>
+        AssumedValid,
+
+        /// <summary>
+        /// Validated using all rules that don't require db state changes.
+        /// </summary>
+        PartiallyValidated,
+
+        /// <summary>
+        /// Validated using all the rules.
```

Comment:
Same problem with assume valid here – this state actually represents that we treat this block as if it was validated using all the rules, although in reality it might not be validated because the rules were skipped.

Reply:
For a second I was thinking whether partially validated and fully validated should not be renamed to partially trusted and fully trusted because the node itself may have not actually validated the blocks. However, then I realised that the validation was actually done, just not by the node itself, but someone else – some other node or a developer – and the owner of the node declared that the validation was done.

So I'm happy with the naming, and I still believe we just need to describe the actual meaning in more detail and then we are fine.